x/masterchef/keeper/hooks_masterchef.go

```diff
@@ -44,8 +44,7 @@ func (k Keeper) UpdateAccPerShare(ctx sdk.Context, poolId uint64, rewardDenom st
 		return
 	}
 	poolRewardInfo.PoolAccRewardPerShare = poolRewardInfo.PoolAccRewardPerShare.Add(
-		math.LegacyNewDecFromInt(amount.Mul(ammtypes.OneShare)).
-			Quo(math.LegacyNewDecFromInt(totalCommit)),
+		math.LegacyNewDecFromInt(amount.Mul(ammtypes.OneShare)).QuoInt(totalCommit),
```

Comment:
@jelysn can you check that `totalCommit` will never be zero to avoid a division by zero panic